x-pack/plugins/security_solution/public/management/pages/policy/store/policy_details/reducer/trusted_apps_reducer.test.ts

```diff
@@ -0,0 +1,264 @@
+import {
+  createLoadedResourceState,
+  createUninitialisedResourceState,
+  createLoadingResourceState,
+  createFailedResourceState,
+} from '../../../../../state';
+import { getListResponse, getAPIError, getFakeCreateResponse } from '../../../test_utils';
+
+describe('policy trusted apps reducer', () => {
+  let initialState: ImmutableObject<PolicyDetailsState>;
+
+  beforeEach(() => {
+    initialState = initialPolicyDetailsState();
+  });
+
+  describe('PolicyTrustedApps', () => {
+    describe('policyArtifactsAvailableListPageDataChanged', () => {
+      it('sets available list uninitialised', () => {
+        const result = policyTrustedAppsReducer(initialState, {
+          type: 'policyArtifactsAvailableListPageDataChanged',
+          payload: createUninitialisedResourceState(),
+        });
+
+        expect(result).toStrictEqual({
+          ...initialState,
+          artifacts: {
+            ...initialState.artifacts,
+            availableList: {
+              type: 'UninitialisedResourceState',
+            },
+          },
+        });
+      });
+      it('sets available list loading', () => {
+        const result = policyTrustedAppsReducer(initialState, {
+          type: 'policyArtifactsAvailableListPageDataChanged',
+          payload: createLoadingResourceState(createUninitialisedResourceState()),
+        });
+
+        expect(result).toStrictEqual({
+          ...initialState,
+          artifacts: {
+            ...initialState.artifacts,
+            availableList: {
+              previousState: {
+                type: 'UninitialisedResourceState',
+              },
+              type: 'LoadingResourceState',
+            },
+          },
+        });
+      });
+      it('sets available list loaded', () => {
+        const result = policyTrustedAppsReducer(initialState, {
+          type: 'policyArtifactsAvailableListPageDataChanged',
+          payload: createLoadedResourceState(getListResponse()),
+        });
+
+        expect(result).toStrictEqual({
+          ...initialState,
+          artifacts: {
+            ...initialState.artifacts,
+            availableList: {
+              data: getListResponse(),
+              type: 'LoadedResourceState',
+            },
+          },
+        });
+      });
+      it('sets available list failed', () => {
+        const result = policyTrustedAppsReducer(initialState, {
+          type: 'policyArtifactsAvailableListPageDataChanged',
+          payload: createFailedResourceState(getAPIError()),
+        });
+
+        expect(result).toStrictEqual({
+          ...initialState,
+          artifacts: {
+            ...initialState.artifacts,
+            availableList: {
+              type: 'FailedResourceState',
+              error: getAPIError(),
+              lastLoadedState: undefined,
+            },
+          },
+        });
+      });
+    });
+  });
+
+  describe('policyArtifactsUpdateTrustedAppsChanged', () => {
```

Comment:
by seeing the reducer code that mostly doesn't contain any business logic and simply alters the state, these tests are testing an implementation detail that is already covered by redux inner tests and the react component tests.

Reply:
Yeah, this is just to ensure that any change on the reducer will make this test fails, so if someone change the resolver result then this will need to be changed.